contracts/drop/DropERC721.sol

```diff
@@ -12,75 +12,43 @@ pragma solidity ^0.8.11;
+import "../dynamic-contracts/extension/Permissions.sol";
+import "../dynamic-contracts/extension/ERC2771ContextUpgradeable.sol";
 
-//  ==========  Internal imports    ==========
-
-import "../openzeppelin-presets/metatx/ERC2771ContextUpgradeable.sol";
-import "../lib/CurrencyTransferLib.sol";
-
-//  ==========  Features    ==========
-
-import "../extension/ContractMetadata.sol";
-import "../extension/PlatformFee.sol";
-import "../extension/Royalty.sol";
-import "../extension/PrimarySale.sol";
-import "../extension/Ownable.sol";
-import "../extension/DelayedReveal.sol";
-import "../extension/LazyMint.sol";
-import "../extension/PermissionsEnumerable.sol";
-import "../extension/Drop.sol";
-
-// OpenSea operator filter
-import "../extension/DefaultOperatorFiltererUpgradeable.sol";
+import "../dynamic-contracts/init/ContractMetadataInit.sol";
+import "../dynamic-contracts/init/PlatformFeeInit.sol";
+import "../dynamic-contracts/init/RoyaltyInit.sol";
+import "../dynamic-contracts/init/PrimarySaleInit.sol";
+import "../dynamic-contracts/init/OwnableInit.sol";
+import "../dynamic-contracts/init/ERC721AInit.sol";
+import "../dynamic-contracts/init/PermissionsEnumerableInit.sol";
+import "../dynamic-contracts/init/DefaultOperatorFiltererInit.sol";
+import "../dynamic-contracts/init/ReentrancyGuardInit.sol";
 
 contract DropERC721 is
     Initializable,
-    ContractMetadata,
-    PlatformFee,
-    Royalty,
-    PrimarySale,
-    Ownable,
-    DelayedReveal,
-    LazyMint,
-    PermissionsEnumerable,
-    Drop,
+    Multicall,
     ERC2771ContextUpgradeable,
-    MulticallUpgradeable,
-    DefaultOperatorFiltererUpgradeable,
-    ERC721AUpgradeable
+    BaseRouter,
+    DefaultOperatorFiltererInit,
+    ContractMetadataInit,
+    PlatformFeeInit,
+    RoyaltyInit,
+    PrimarySaleInit,
+    OwnableInit,
+    PermissionsEnumerableInit,
+    ERC721AInit
 {
-    using StringsUpgradeable for uint256;
-
-    /*///////////////////////////////////////////////////////////////
-                            State variables
-    //////////////////////////////////////////////////////////////*/
-
-    /// @dev Only transfers to or from TRANSFER_ROLE holders are valid, when transfers are restricted.
-    bytes32 private transferRole;
-    /// @dev Only MINTER_ROLE holders can sign off on `MintRequest`s and lazy mint tokens.
-    bytes32 private minterRole;
-
-    /// @dev Max bps in the thirdweb system.
-    uint256 private constant MAX_BPS = 10_000;
-
-    /// @dev Global max total supply of NFTs.
-    uint256 public maxTotalSupply;
-
-    /// @dev Emitted when the global max supply of tokens is updated.
-    event MaxTotalSupplyUpdated(uint256 maxTotalSupply);
-
     /*///////////////////////////////////////////////////////////////
                     Constructor + initializer logic
     //////////////////////////////////////////////////////////////*/
 
-    constructor() initializer {}
+    constructor(Extension[] memory _extensions) BaseRouter(_extensions) {}
```

Comment:
make a new contract for this, don't touch the existing contracts in case we need to ship fixes